src/client/java/hadoop-daos/src/main/java/io/daos/fs/hadoop/DaosFileSource.java

```diff
@@ -38,24 +38,30 @@ public abstract class DaosFileSource {
 
   private static final Logger LOG = LoggerFactory.getLogger(DaosFileSource.class);
 
-  protected DaosFileSource(DaosFile daosFile, int bufCapacity, long fileLen,
+  protected DaosFileSource(DaosFile daosFile, int bufCapacity, long fileLen, boolean append,
                            FileSystem.Statistics stats) {
     this.daosFile = daosFile;
     this.buffer = BufferAllocator.directNettyBuf(bufCapacity);
     selfManagedBuf = true;
     this.bufCapacity = buffer.capacity();
     this.fileLen = fileLen;
     this.stats = stats;
+    if (append) {
```

Comment:
do we need to require multithreaded access with the same handle?
like what if two files are appending to the same file at the same time? should this be serialized with some lock?

Reply:
> do we need to require multithreaded access with the same handle? like what if two files are appending to the same file at the same time? should this be serialized with some lock?

I checked the append impl in HDFS, which is Hadoop's impl of Hadoop Filesystem. HDFS has a concept, file lease, in namenode which prevents concurrent update operations to same file. As talked in last meeting, DAOS doesn't have similar control to file. It seems that there is no API in client to lock a file.

So for now, I just assume there is no more 1 thread updating same file.